src/dotnet/APIView/APIViewWeb/Managers/ReviewManager.cs

```diff
@@ -144,6 +160,45 @@ public async Task<LegacyReviewModel> GetLegacyReviewAsync(ClaimsPrincipal user,
             return review;
         }
 
+        /// <summary>
+        /// Get Review if it exist otherwise create it
+        /// </summary>
+        /// <param name="file"></param>
+        /// <param name="filePath"></param>
+        /// <param name="language"></param>
+        /// <param name="runAnalysis"></param>
+        /// <returns></returns>
+        public async Task<ReviewListItemModel> GetOrCreateReview(IFormFile file, string filePath, string language, bool runAnalysis = false)
```

Comment:
Where is this API triggered from? I see that you are processing incoming file stream only to get package name and language. I assume that caller of this API processes file stream again to create and store actual code file when creating revision within the review. Can we process file stream and create memory stream only once and pass down package name and language to this method if that's the case?